### ISSUE TITLE
convox uninstall always fails with a ValidationError for an S3 resource

### ISSUE THREAD

**Issue by ndbroadbent** [convox/rack#2701]
I can reproduce this every time I uninstall my test rack. Here's the full logs:

```
$ convox uninstall my-app us-east-1


     ___    ___     ___   __  __    ___   __  _
    / ___\ / __ \ /  _  \/\ \/\ \  / __ \/\ \/ \
   /\ \__//\ \_\ \/\ \/\ \ \ \_/ |/\ \_\ \/>  </
   \ \____\ \____/\ \_\ \_\ \___/ \ \____//\_/\_\
    \/____/\/___/  \/_/\/_/\/__/   \/___/ \//\/_/


Resources to delete:
STACK                    TYPE      STATUS
appname-uploads-346a0a39b8  resource  CREATE_COMPLETE
redis-7621               resource  CREATE_COMPLETE
postgres-3615            resource  CREATE_COMPLETE
appname                  app       UPDATE_COMPLETE
my-app       rack      UPDATE_COMPLETE

Delete everything? y/N: y

Uninstalling Convox...
Deleting appname-uploads-346a0a39b8...
Deleting redis-7621...
Deleting postgres-3615...
Deleting appname...
Deleted Unknown: AWS::ApplicationAutoScaling::ScalableTarget: service/my-app-Cluster-1EQBGNSIOIAB6/my-app-appname-ServiceWorkerService-16VD3PKL11S50|ecs:service:DesiredCount|ecs
Deleted Unknown: AWS::ElasticLoadBalancingV2::ListenerRule: BalancerWebListenerRule443
Deleted Unknown: AWS::ApplicationAutoScaling::ScalingPolicy: ServiceWorkerAutoscalingPolicyMemory
Deleted Unknown: AWS::ApplicationAutoScaling::ScalingPolicy: ServiceWorkerAutoscalingPolicyCpu
Deleted Unknown: AWS::ApplicationAutoScaling::ScalingPolicy: ServiceWebAutoscalingPolicyMemory
Deleted Unknown: AWS::ApplicationAutoScaling::ScalingPolicy: ServiceWebAutoscalingPolicyRequests
Deleted Unknown: AWS::ApplicationAutoScaling::ScalingPolicy: ServiceWebAutoscalingPolicyCpu
Deleted Unknown: AWS::ElasticLoadBalancingV2::ListenerRule: BalancerWebListenerRule443Internal
Deleted Unknown: AWS::ElasticLoadBalancingV2::ListenerRule: BalancerWebListenerRule80Internal
Deleted ECS TaskDefinition: ServiceWorkerTasks
Deleted ECS TaskDefinition: ServiceWebTasks
Deleted ECS TaskDefinition: TimerSubmissionExpirationTasks
Deleted Unknown: AWS::ApplicationAutoScaling::ScalableTarget: service/my-app-Cluster-1EQBGNSIOIAB6/my-app-appname-ServiceWebService-1DULPSZWJCF8|ecs:service:DesiredCount|ecs
Deleted Lambda Function: my-app-appname-TimerLauncher-1CV1ZPS52YRR7
Deleted ECS TaskDefinition: ServiceWorkerTasks
Deleted ECS Service: ServiceWorkerService
Deleted ECS TaskDefinition: TimerSubmissionExpirationTasks
Uninstalling Convox...
Deleted CloudWatch Log Group: my-app-appname-LogGroup-1VFHOIT11A7SU
Skipped Unknown: AWS::ECR::Repository: myappa-regis-6euzlhtmgmd8
Deleted ECS TaskDefinition: ServiceWebTasks
Deleted Unknown: AWS::ElasticLoadBalancingV2::ListenerRule: BalancerWebListenerRule80
Deleted ECS Service: ServiceWebService
Deleting my-app...
Deleted ECS Cluster: my-app-BuildCluster-1GCTL4H4CK9NL
Deleted AutoScalingGroup: my-app-BuildInstances-1JMP9WDLRRT99
Deleted ECS TaskDefinition: ApiWebTasks
Deleted ECS TaskDefinition: ApiBuildTasks
Deleted ECS TaskDefinition: ApiMonitorTasks
Deleted ECS TaskDefinition: ApiBuildTasks
Deleted Routing Table: rtb-c2fd0bbd
Deleted Lambda Function: my-app-InstancesLifecycleHandler-PP9KUKFCUMTD
Deleted Security Group: sg-7a346132
Deleted ECS TaskDefinition: ApiMonitorTasks
Deleted ECS Service: ApiMonitorService
Uninstalling Convox...
Deleted ECS TaskDefinition: ApiWebTasks
Deleted ECS Service: ApiWebService
Deleted KMS Alias: alias/convox-my-app
Deleted KMS Key: EncryptionKey
Deleted Elastic Load Balancer: my-app
Deleted CloudWatch Log Group: my-app-LogGroup-15EHIKZ7N9W2W
Skipped S3 Bucket: my-app-settings-1cis5w8i82cxl
Deleted Application Load Balancer: Router
Deleted SSL Certificate: RouterApiCertificate
Skipped S3 Bucket: my-app-logs-1cjpk3nhynlpr
Deleted S3 Bucket Policy: my-app-LogsPolicy-T9C1OP3LKQ8A
Deleted DynamoDB Table: my-app-releases
Deleted DynamoDB Table: my-app-builds
Deleted Hosted Zone: Z3FIXXKDJRUZFV
Deleted SQS Queue: https://sqs.us-east-1.amazonaws.com/***********/my-app-CloudmyappationEvents-GSW7BK907HO2
Deleted SQS Queue: https://sqs.us-east-1.amazonaws.com/***********/my-app-AccountEvents-R0P5VSOI0QX9
Deleted Security Group: sg-40306508
Uninstalling Convox...
Deleted Security Group: sg-98396cd0
openssl rand -hex 64Uninstalling Convox...
Deleted Security Group: sg-21326769
Deleted ECS Cluster: my-app-Cluster-1EQBGNSIOIAB6
Deleted AutoScalingGroup: my-app-Instances-1RHCXV47SQVLU
Deleted Lambda Function: my-app-CustomTopic-1X5UKG6537E0V
Deleted VPC Subnet: subnet-ec0ac18b
Deleted VPC Subnet: subnet-c80fcd94
Deleted VPC Subnet: subnet-79b5ad46
Deleted VPC Internet Gateway: igw-2608455e
Deleted EFS Filesystem: fs-3d6cee75
Emptying S3 Bucket my-app-appname-uploads-346a0a39b8...
Emptying S3 Bucket my-app-logs-1cjpk3nhynlpr...
Deleting S3 Bucket my-app-appname-uploads-346a0a39b8...
Emptying S3 Bucket my-app-settings-1cis5w8i82cxl...
Deleting S3 Bucket my-app-logs-1cjpk3nhynlpr...
Deleting S3 Bucket my-app-settings-1cis5w8i82cxl...
Emptying S3 Bucket my-app-appname-settings-1wx4qmi57o8ha...
Deleting S3 Bucket my-app-appname-settings-1wx4qmi57o8ha...
ERROR: ValidationError: Stack [appname-uploads-346a0a39b8] does not exist
	status code: 400, request id: c5dfac1c-6325-11e8-9da9-35de48ef6856
ERROR: Uninstall encountered some errors, contact support@convox.com for assistance

```

**Comment by ndbroadbent** [comment]
It looks like it deletes `appname-uploads-346a0a39b8` at the very beginning, then tries to delete it again before crashing.

**Comment by ndbroadbent** [comment]
I attempted to fix this in https://github.com/FormAPI/convox-rack/commit/0dda546c846bbf05832de041f6989fab03dc0681 (reorder the operations), but that didn't work.

**Comment by ddollar** [comment]
This is fixed in #2379
